classes/file/FileManager.inc.php

```diff
@@ -348,7 +348,11 @@ function rmtree($file) {
 	 */
 	function mkdirtree($dirPath, $perms = null) {
 		if (!file_exists($dirPath)) {
-			if ($this->mkdirtree(dirname($dirPath), $perms)) {
+			//Avoid infinite recursion when file_exists reports false for root directory
+			if ($dirPath == dirname ( $dirPath )) {
+				trigger_error ( 'There are no readable files in this directory tree. Is safe mode or open_basedir active?', E_USER_ERROR );
```

Comment:
Thanks for contributing, @pmarrone! Two minor changes to suggest for this line:
- Could you change `trigger_error` to `fatalError` for consistency with the rest of the codebase?
- Just a typo, and sorry for being pedantic: "Is safe mode or open_basedir active?" should be "_Are_ safe mode or open_basedir active?"